content/en/blog/_posts/2023-12-15-volume-attributes-class/index.md

```diff
@@ -0,0 +1,179 @@
+the external-resizer sidecar will get an informer event. Based on the current state of the configuration,
+the resizer will trigger a CSI ControllerModifyVolume.
+More details can be found in [KEP-3751](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/3751-volume-attributes-class/README.md).
+
+## How to use it
+
+If you want to test the feature whilst it's alpha, you need to enable the relevant feature gate
+in the `kube-controller-manager` and the `kube-apiserver`. Use the `--feature-gates` command line argument:
+
+
+```
+--feature-gates="...,VolumeAttributesClass=true"
+```
+
+
+It also requires that the CSI driver has implemented the ModifyVolume API.
+
+
+### User Flow
+
+If you would like to see the feature in action and verify it works fine in your cluster, here's what you can try:
+
+
+1. Define a StorageClass and VolumeAttributesClass
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: csi-sc-example
+provisioner: pd.csi.storage.gke.io
+parameters:
+  disk-type: "hyperdisk-balanced"
+volumeBindingMode: WaitForFirstConsumer
+```
+
+
+
+```yaml
+apiVersion: storage.k8s.io/v1alpha1
+kind: VolumeAttributesClass
+metadata:
+  name: silver
+driverName: pd.csi.storage.gke.io
+parameters:
+  provisioned-iops: "3000"
+  provisioned-throughput: "50"
+```
+
+
+2. Define and create the PersistentVolumeClaim
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: test-pv-claim
+spec:
+  storageClassName: csi-sc-example
+  volumeAttributesClassName: silver
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 64Gi
+```
+
+
+3. Verify that the PersistentVolumeClaim is now provisioned correctly with:
+
+```yaml
+kubectl get pvc
+```
+
+
+4. Create a new VolumeAttributesClass gold:
+
+```yaml
+apiVersion: storage.k8s.io/v1alpha1
+kind: VolumeAttributesClass
+metadata:
+  name: gold
+driverName: pd.csi.storage.gke.io
+parameters:
+  iops: "4000"
+  throughput: "60"
+```
+
+
+5. Update the PVC with the new VolumeAttributesClass and apply:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: test-pv-claim
+spec:
+  storageClassName: csi-sc-example
+  volumeAttributesClassName: gold
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 64Gi
+
+```
+
+
+6. Verify that PersistentVolumeClaims has the updated VolumeAttributesClass parameters with:
+
+```yaml
+kubectl describe pvc <PVC_NAME>
+```
+
+## Next steps
+
+* See the [VolumeAttributesClass KEP](https://kep.k8s.io/3751) for more information on the design
+* You can view or comment on the [project board](https://github.com/orgs/kubernetes-csi/projects/72) for VolumeAttributesClass
+* In order to move this feature towards beta, we need feedback from the community,
+so here's a call to action: add support to the CSI drivers, try out this feature,
+consider how it can help with problems that your users are having…
```

Comment:
```suggestion
  so here's a call to action: add support to the CSI drivers, try out this feature,
  consider how it can help with problems that your users are having…
```